### PR TITLE
Gowtham katta

### PR DESCRIPTION
**Pull request by gothamcity17**
Merge branch gowtham-katta with branch master

### CODE REVIEW

File: oop1-construct.js [hunk 1]
```diff
@@ -2,105 +2,323 @@
...
 //your code here...
 
+class Amazon {
+  constructor(name, classification, color, actions) {
+    this.name = name;
+    this.classification = classification;
+    this.color = color;
+    this.actions = actions;
+  }
+  z;
```
Comment (tek-abrjablonski) [inline oop1-construct.js:39]: typo on line 39?

File: oop1-construct.js [hunk 2]
```diff
@@ -2,105 +2,323 @@
...
+      this.name = name;
+      this.sides = sides;
+      this.base = base;
+      this.height = height;
+      this.length = length;
+      this.width = width;
+      this.radius = radius;
+  }
+
+  calcArea() {
```
Comment (tek-abrjablonski) [inline oop1-construct.js:207]: missing explanations of the formulas - these are just output of a calculated variable

File: oop1-construct.js [hunk 3]
```diff
@@ -2,105 +2,323 @@
...
+octagon.calcArea();
+octagon.calcPerimeter();
+
+
+/*
 //5. Below is a class Earth.  It is instantiated with a earth 'instance'.  Because there is only 1 known earth, we don't want multiple instances of earth.  Modify the class so the properties are usuable WITHOUT instantiating the class.
 
 */
 
-class Earth {
```
Comment (tek-abrjablonski) [inline oop1-construct.js:90]: How are you preventing multiple instances?

File: oop2-encapsulate-inherit.js [hunk 4]
```diff
@@ -22,24 +122,135 @@
     
     - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
 
+    //your code here...
```
Comment (tek-abrjablonski) [inline oop2-encapsulate-inherit.js:125]: too much inheritance here...  A Site or Employee should not be an Umbrella.  Also, your constructor chaining is mismatched: you're calling super with parameter sets that don't have matching constructors

Comment (tek-abrjablonski) [reply oop2-encapsulate-inherit.js:125]: reviewing the instructions, it's actually requesting this hierarchy, so just keep it in mind for the future (has-a vs is-a relationships)

File: oop1-construct.js [hunk 5]
```diff
@@ -2,105 +2,323 @@
...
+    console.log(`${this.name}'s perimeter is calculated to be : ${2 * this.length + 2 * this.width}`);
+  },
+};
 
 console.log(rectangle);
 rectangle.calcArea();
 rectangle.calcPerimeter();
 
+
+const newCircle = new Shape("circle")
```
Comment (mbennett5) [inline oop1-construct.js:172]: This line causes an error to be thrown because newCircle has already been declared on line 124

**Review by mbennett5 — APPROVED** [review]

**Pull request merged**